# This is only required for flatpak
# In the sandbox - the programs in the Exec and Try_Exec keys do not exist
# it's not a problem when launching the app since we use flatpak-spawn
# but when listing the apps - GLIb will filter them out

# https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/blob/master/elements/components/glib.bst
# https://gitlab.gnome.org/GNOME/gnome-build-meta/-/blob/master/elements/sdk/glib.bst

diff --git a/gio/gdesktopappinfo.c b/gio/gdesktopappinfo.c
index 63ef0c045..fa5ade90a 100644
--- a/gio/gdesktopappinfo.c
+++ b/gio/gdesktopappinfo.c
@@ -1787,47 +1787,11 @@ g_desktop_app_info_load_from_keyfile (GDesktopAppInfo *info,
                                     G_KEY_FILE_DESKTOP_GROUP,
                                     G_KEY_FILE_DESKTOP_KEY_TRY_EXEC,
                                     NULL);
-  if (try_exec && try_exec[0] != '\0')
-    {
-      char *t;
-      t = g_find_program_in_path (try_exec);
-      if (t == NULL)
-        {
-          g_free (try_exec);
-          return FALSE;
-        }
-      g_free (t);
-    }

   exec = g_key_file_get_string (key_file,
                                 G_KEY_FILE_DESKTOP_GROUP,
                                 G_KEY_FILE_DESKTOP_KEY_EXEC,
                                 NULL);
-  if (exec && exec[0] != '\0')
-    {
-      gint argc;
-      char **argv;
-      if (!g_shell_parse_argv (exec, &argc, &argv, NULL))
-        {
-          g_free (exec);
-          g_free (try_exec);
-          return FALSE;
-        }
-      else
-        {
-          char *t;
-          t = g_find_program_in_path (argv[0]);
-          g_strfreev (argv);
-
-          if (t == NULL)
-            {
-              g_free (exec);
-              g_free (try_exec);
-              return FALSE;
-            }
-          g_free (t);
-        }
-    }

   info->name = g_key_file_get_locale_string (key_file, G_KEY_FILE_DESKTOP_GROUP, G_KEY_FILE_DESKTOP_KEY_NAME, NULL, NULL);
   info->generic_name = g_key_file_get_locale_string (key_file, G_KEY_FILE_DESKTOP_GROUP, GENERIC_NAME_KEY, NULL, NULL);
